src/java/htsjdk/samtools/SAMSequenceDictionary.java

```diff
@@ -163,8 +165,72 @@ public boolean equals(Object o) {
         return true;
     }
 
+    /** return a MD5 sum for ths dictionary, the checksum is not re-computed each time this
+     * method is called. <pre>md5( (seq1.name+seq1.length) + ' '+(seq2.name+seq2.length) + ' '+...)</pre>
+     * @return a MD5 checksum for this dictionary or the empty string if it is empty
+     */
+    public String md5() {
+        if(isEmpty()) return "";
+        try {
+            MessageDigest md5 = MessageDigest.getInstance("MD5");
+            md5.reset();
+            for(SAMSequenceRecord ssr : mSequences)  {
+                if(ssr.getSequenceIndex()>0)  md5.update((byte)' ');
+                md5.update(ssr.getSequenceName().getBytes());
+                md5.update(String.valueOf(ssr.getSequenceLength()).getBytes());
```

Comment:
Perhaps we also use the ```MD5_TAG``` as well if it is available?  This would really help in the case that the sequence names and lengths are the same, but the bases differ.  It happens, seriously.